src/main/java/org/traccar/protocol/Gt06ProtocolDecoder.java

```diff
@@ -779,16 +788,20 @@ private Object decodeBasicOther(
             }
 
         } else if (type == MSG_ALARM) {
-
             DateBuilder dateBuilder = new DateBuilder(deviceSession.getTimeZone())
                     .setDate(buf.readUnsignedByte(), buf.readUnsignedByte(), buf.readUnsignedByte())
                     .setTime(buf.readUnsignedByte(), buf.readUnsignedByte(), buf.readUnsignedByte());
-
-            getLastLocation(position, dateBuilder.getDate());
-
+            if (dataLength == 7) {
```

Comment:
Is this a way to identify device model? Maybe we can have a boolean variable for it? So it's more readable.

Reply:
Not a nice solution, is it?

JC100 sends this packet for harsh braking: `78780c95130a0209321c91000112800d0a`

Same packet for JC400: `78788b95140a060e2208055D4A800209D9C014c59100004556454e545f3335333337363131303032333139365f30303030303030305f323032305f31305f30365f31365f33345f30385f33322e6d70342c4556454e545f3335333337363131303032333139365f30303030303030305f323032305f31305f30365f31365f33345f30385f33312e6d70340000da360d0a`

The long part of JC400 message starting with `455645...` are names of uploaded video files - the device does that the moment an event happens.

So the important parts of both packets, stripped of header/length/type:

```
130a0209321c 91
140a060e2208 055D4A80 0209D9C0 14c5 91
```

It is date/time the the beginning for both, JC400 continues with GPS location and event type while the JC100 has only the event type there.

The packet length was the obvious solution for me here. Any idea of a better solution?

Reply:
I'm ok with the solution. I think we just need add a variable to make it more explicit.